packages/relay/src/services/ReportService.ts

```diff
@@ -280,6 +295,389 @@ export class ReportService {
             },
         ]
     }
+
+    async claimPositiveReputation(
+        req: express.Request,
+        res: express.Response,
+        db: DB
+    ) {
+        await this.claimReputation(
+            req,
+            res,
+            db,
+            ClaimMethods.CLAIM_POSITIVE_REP,
+            ReputationDirection.POSITIVE
+        )
+    }
+
+    async claimNegativeReputation(
+        req: express.Request,
+        res: express.Response,
+        db: DB
+    ) {
+        await this.claimReputation(
+            req,
+            res,
+            db,
+            ClaimMethods.CLAIM_NEGATIVE_REP,
+            ReputationDirection.NEGATIVE
+        )
+    }
+
+    async claimReputation(
```

Comment:
It is necessary to check if epockKey in proof is equal to the epockKey in ReportHistory.

Reply:
just check VOTER's  reportNullifier, poster and repotrer can't get  epockKey from publicSignals

Reply:
fixed